File: Sorting and Searching/Playlist/playlist.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class playlist {
    private static void solve(){
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] arr = new int[n];
        for(int i=0;i<n;i++){
            arr[i] = scanner.nextInt();
        }
        int l=0, r=0, ans=0;
        Map<Integer, Integer> mp = new HashMap<>();
        while(r<n){
            if(mp.containsKey(arr[r]) == true){
                l = Math.max(l, mp.get(arr[r])+1);
            }
            mp.put(arr[r], r);
            ans = Math.max(ans, r-l+1);
            r++;
        }
        System.out.println(ans);

    }
    public static void main(String args[]){
        solve();
    }    
}

File: Sorting and Searching/Playlist/playlist.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long int

void solve(){
	int n, ans = 0;
	cin>>n;
	vector<int> arr(n);
	map<int, int> mp;
	
	for(int i=0;i<n;i++){
		cin>>arr[i];
	}

	int l=0, r=0;
	while(r<n){
		if(mp.find(arr[r]) != mp.end()){
			l = max(l, mp[arr[r]]+1);
		}
		mp[arr[r]] = r;
		ans = max(ans, r-l+1);
		r++;
	}
	cout<<ans;
    return;
	
}

int main(){
	long long testCases;
	testCases = 1;
	// cin>>testCases;
	while(testCases--){
		solve();
	}
	return 0;
}